### PR TITLE
independent decompression test

### PR DESCRIPTION
**Pull request by michalek-no**
Stand alone tool that extracts compressed stream from signed.bin, decompresses it and compares against uncompressed application binary.

### CODE REVIEW

**Comment by nordicjm** [comment]
> > bit confused as to why this wouldn't be done using zephyr with ztest
> 
> we wanted to have something external, also to show how to handle it without all the machinery that comes with zephyr and ncs

Important to note that this is very linux specific, this will not compile or work on windows and doubt it would on mac either

**Comment by michalek-no** [comment]
> > > bit confused as to why this wouldn't be done using zephyr with ztest
> > 
> > 
> > we wanted to have something external, also to show how to handle it without all the machinery that comes with zephyr and ncs
> 
> Important to note that this is very linux specific, this will not compile or work on windows and doubt it would on mac either

good :)

**Review by nvlsianpu — COMMENTED** [review]
I like the idea of testing that imgtool.py feature using independent from NCS support thing -  that exactly what this code allows. Let give voice to the verification on how we can introduce CI test using this.

File: tests/subsys/nrf_compress/decompression/independent_cmp.c
Comment (gchwier) [inline tests/subsys/nrf_compress/decompression/independent_cmp.c:1]: it should be rather cpp file, not C.
When I try to compile it with gcc I receive errors (unless I tell the compiler, it is C++ standard),
Compilation works ok with g++.

**Review by gchwier — COMMENTED** [review]
Please write any readme file and explain how to use that tool (how to compile etc).
Maybe could be worth to investigate extending imgtool (in mcuboot repo) with verify method for LZMA?

**Comment by michalek-no** [comment]
rebase

**Comment by michalek-no** [comment]
rebase

**Comment by michalek-no** [comment]
rebase

**Review by annwoj — APPROVED** [review]

**Pull request closed without merging**